### PR TITLE
[BACKLOG-14992] Fix to conversions from String when a mask isn't already set.

### PR DESCRIPTION
**Pull request by davidmsantos90**
@pentaho/millenniumfalcon @TatsianaKasiankova please review

The second commit is a cleanup (removing deprecated Assert.* ) and it is the source of most of changed lines

### CODE REVIEW

**Comment by nantunes** [comment]
@TatsianaKasiankova sorry for merging so quickly, didn't notice you were mentioned. Please review and comment if you find something, this can be reverted if needed.

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 1]
```diff
@@ -80,27 +80,45 @@
...
-  //TODO: this value was extracted from the code. Does it make sense to have the same mask for this types?
-  static final String DEFAULT_NUMBER_FORMAT_MASK = " ##########0.0########;-#########0.0########";
-  static final String DEFAULT_BIG_NUMBER_FORMAT_MASK = " ##########0.0########;-#########0.0########";
+  static final String DEFAULT_BIG_NUMBER_FORMAT_MASK = Const.NVL(
+          EnvUtil.getSystemProperty( Const.KETTLE_DEFAULT_BIGNUMBER_FORMAT ),
+          "######0.0###################;-######0.0###################" );
 
-  public static final String DEFAULT_DATE_FORMAT_MASK = Const.NVL( EnvUtil
-      .getSystemProperty( Const.KETTLE_DEFAULT_DATE_FORMAT ), "yyyy/MM/dd HH:mm:ss.SSS" );
+  static final String DEFAULT_DATE_FORMAT_MASK = Const.NVL( EnvUtil
```
Comment (TatsianaKasiankova) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: Changed access modifier - but this constant could be used in the other projects.
E.g. here - https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/org/pentaho/di/repository/kdr/delegates/KettleDatabaseRepositoryMetaStoreDelegate.java#L393

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 2]
```diff
@@ -80,27 +80,45 @@
...
+          .getSystemProperty( Const.KETTLE_DEFAULT_DATE_FORMAT ), "yyyy/MM/dd HH:mm:ss.SSS" );
 
-  public static final String DEFAULT_TIMESTAMP_FORMAT_MASK = Const.NVL( EnvUtil
-      .getSystemProperty( Const.KETTLE_DEFAULT_TIMESTAMP_FORMAT ), "yyyy/MM/dd HH:mm:ss.SSSSSSSSS" );
+  static final String DEFAULT_TIMESTAMP_FORMAT_MASK = Const.NVL( EnvUtil
```
Comment (TatsianaKasiankova) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: The same as above - changed access modifier.

Comment (TatsianaKasiankova) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: @davidmsantos90 , @nantunes I've noticed a couple of potential issues. Please take a look on the comments above.

Comment (nantunes) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: @TatsianaKasiankova Thanks. It shows how important your review was... as it did break the QAT. :-(

It's been fixed by https://github.com/pentaho/pentaho-kettle/pull/3651.

**Comment by luckyxiang2017** [comment]
the default mask is shit,i cannot update from 6.1 to 8.0,because the big number has .0

the modify is faild, not good for big number,and has no reason to modify the default mask

**Pull request merged**